agreement_rebate/models/agreement.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2020 Tecnativa - Carlos Dauden
+# Copyright 2020 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, models, fields
+
+
+class Agreement(models.Model):
+    _inherit = 'agreement'
+
+    def _selection_rebate_type(self):
+        return [
+            ('global', 'Global (A discount global for all lines)'),
+            ('line', 'Line (A discount for every line)'),
+            ('section_total', 'Compute total and apply discount rule match'),
+            ('section_prorated', 'Compute multi-dicounts by sections amount'),
+        ]
+
+    rebate_type = fields.Selection(
+        selection=lambda self: self._selection_rebate_type(),
+        string='rebate type',
+    )
+    rebate_line_ids = fields.One2many(
+        comodel_name='agreement.rebate.line',
+        string='Agreement rebate lines',
+        inverse_name='agreement_id',
+        copy=True,
+    )
+    rebate_section_ids = fields.One2many(
+        comodel_name='agreement.rebate.section',
+        string='Agreement rebate sections',
+        inverse_name='agreement_id',
+        copy=True,
+    )
+    rebate_discount = fields.Float(
+        string='Rebate discount',
+    )
+
+
+class AgreementRebateLine(models.Model):
+    _name = 'agreement.rebate.line'
+    _description = 'Agreement Rebate Lines'
+
+    agreement_id = fields.Many2one(
+        comodel_name='agreement',
+        string='Agreement',
+    )
+    rebate_target = fields.Selection([
+        ('product', 'Product variant'),
+        ('product_tmpl', 'Product templates'),
+        ('category', 'Product categories'),
+        ('condition', 'Rebate condition'),
+        ('domain', 'Rebate domain'),
+    ])
+    rebate_product_ids = fields.Many2many(
```

Comment:
Same argument for the rest of the fields

```suggestion
    product_ids = fields.Many2many(
```